oracle-price-monitor/src/autotask.js

```diff
@@ -0,0 +1,270 @@
+      throw error;
+    }
+  }
+
+  return response;
+}
+
+async function getFortaAlerts(agentId, transactionHash) {
+  const headers = {
+    'content-type': 'application/json',
+  };
+
+  const graphqlQuery = {
+    operationName: 'recentAlerts',
+    query: `query recentAlerts($input: AlertsInput) {
+      alerts(input: $input) {
+        pageInfo {
+          hasNextPage
+          endCursor {
+            alertId
+            blockNumber
+          }
+        }
+        alerts {
+          createdAt
+          name
+          protocol
+          findingType
+          hash
+          source {
+            transactionHash
+            block {
+              number
+              chainId
+            }
+            agent {
+              id
+            }
+          }
+          severity
+		  metadata
+		  description
+        }
+      }
+    }`,
+    variables: {
+      input: {
+        first: 100,
+        agents: [agentId],
+        transactionHash,
+        createdSince: 0,
+        chainId: 1,
+      },
+    },
+  };
+
+  // perform the POST request
+  const response = await axios.post(
+    fortaApiEndpoint,
+    graphqlQuery,
+    headers,
+  );
+
+  const { data } = response;
+  if (data === undefined) {
+    return undefined;
+  }
+
+  console.log('Forta Public API data');
+  console.log(JSON.stringify(data, null, 2));
+  const { data: { alerts: { alerts } } } = data;
+  return alerts;
+}
+
+// entry point for autotask
+// eslint-disable-next-line func-names
+exports.handler = async function (autotaskEvent) {
+  // ensure that the autotaskEvent Object exists
+  if (autotaskEvent === undefined) {
+    return {};
+  }
+  console.log('Autotask Event');
+  console.log(JSON.stringify(autotaskEvent, null, 2));
+
+  const { secrets } = autotaskEvent;
+  if (secrets === undefined) {
+    return {};
+  }
+
+  // ensure that there is a DiscordUrl secret. Name changes depending on what webhook secret you use
+  const { FortaSentinelTestingDiscord: discordUrl } = secrets;
+  if (discordUrl === undefined) {
+    return {};
+  }
+
+  // ensure that the request key exists within the autotaskEvent Object
+  const { request } = autotaskEvent;
+  if (request === undefined) {
+    return {};
+  }
+
+  // ensure that the body key exists within the request Object
+  const { body } = request;
+  if (body === undefined) {
+    return {};
+  }
+  console.log('Body');
+  console.log(JSON.stringify(body, null, 2));
+
+  // ensure that the alert key exists within the body Object
+  const { alert } = body;
+  if (alert === undefined) {
+    return {};
+  }
+
+  // extract the transaction hash and agent ID from the alert Object
+  const {
+    hash,
+    source: {
+      transactionHash,
+      agent: {
+        id: agentId,
+      },
+    },
+  } = alert;
+
+  // retrieve the metadata from the Forta public API
+  let alerts = await getFortaAlerts(agentId, transactionHash);
+  // alerts = alerts.filter((alertObject) => alertObject.hash === hash);
```

Comment:
```suggestion
  const alerts = await getFortaAlerts(agentId, transactionHash);
```
If you don't need this sanity check, then remove it.